Menus/BuyMenuGUI.cpp

```diff
@@ -2233,24 +2253,58 @@ void BuyMenuGUI::AddPresetsToItemList()
 }
 
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          UpdateTotalCostLabel
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Updated the text of the total cost label to reflect the total cost of
-//                  all the items in the order box.
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void BuyMenuGUI::UpdateTotalCostLabel(int whichTeam)
-{
-    char newText[512];
-    std::snprintf(newText, sizeof(newText), "Cost: %.0f/%.0f", GetTotalOrderCost(), g_ActivityMan.GetActivity()->GetTeamFunds(whichTeam));
-    m_pCostLabel->SetText(newText);
+void BuyMenuGUI::UpdateTotalCostLabel(int whichTeam) {
+	// Total order cost is always rounded up from any remainder.
+	std::string display = "Cost: " + RoundFloatToPrecision(GetTotalOrderCost(), 0, 2) + "/" + RoundFloatToPrecision(g_ActivityMan.GetActivity()->GetTeamFunds(whichTeam), 0);
+	m_pCostLabel->SetText(display);
 }
 
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Virtual Method:  TryPurchase
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Attempts to make a purchase with everything set up.
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void BuyMenuGUI::UpdateTotalMassLabel(const ACraft* pCraft, GUILabel* pLabel) const {
+	if (!pLabel) {
+		return;
+	}
+
+	std::string display;
+	if (pCraft && pCraft->GetMaxInventoryMass() != 0) {
+		display = RoundFloatToPrecision(GetTotalOrderMass(), 1, 2);
+		if (pCraft->GetMaxInventoryMass() > 0) {
+			display += " / " + RoundFloatToPrecision(pCraft->GetMaxInventoryMass(), 1);
+		}
+	} else {
+		display = "NO CARGO SPACE";
+	}
+
+	pLabel->SetText(display);
+}
+
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void BuyMenuGUI::UpdateTotalPassengersLabel(const ACraft* pCraft, GUILabel* pLabel) const {
+	if (!pLabel) {
+		return;
+	}
+
+	std::string display;
+	if (pCraft && pCraft->GetMaxPassengers() != 0) {
+		display = std::to_string(GetTotalOrderPassengers());
+		if (pCraft->GetMaxPassengers() > 0) {
+			display += " / " + std::to_string(pCraft->GetMaxPassengers());
+		}
+	} else {
+		display = "NO SPACE";
```

Comment:
Should this say `NO PASSENGER SLOTS` or `NO PASSENGER SPACE`? Maybe it's fine as-is.

Reply:
So this was actually what I had initially, and it looked far better, _but_ ;
The text overlaps the rest of the menu. 
![image](https://user-images.githubusercontent.com/73866773/111907000-5a0c7400-8aa3-11eb-808d-656ab74fca5c.png)
It would look great, but I'm guessing it's gonna be a ton more work to just make the GUI resize as required?
I'd prefer it to have the word "`PASSENGER`" in there as well, what are your thoughts?

Reply:
I'm leaving this for now, with a personal note that if the GUI gets updated to factor it in.